hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -822,10 +839,11 @@ public AbfsRestOperation deletePath(final String path, final boolean recursive,
    * delete issued from this filesystem instance.
    * These are few corner cases and usually returning a success at this stage
    * should help the job to continue.
-   * @param op Delete request REST operation response
+   * @param op Delete request REST operation response with non-null HTTP response
    * @return REST operation response post idempotency check
    */
   public AbfsRestOperation deleteIdempotencyCheckOp(final AbfsRestOperation op) {
+    assert op.hasResult();
```

Comment:
why assert & not Preconditions.checkArgument()? you confident that we don't normally need this check? or happy with the fact that there'll be an NPE shortly afterwards anyway?

Reply:
My thinking was that the callers of this method (and others) would just be internal to hadoop-azure. Just realized that I forgot to add a corresponding assert to `renameIdempotencyCheckOp(AbfsRestOperation)` like I wanted to.

I don't feel strongly that this is `assert` vs. `Preconditions.checkArgument()`. I am happy to make this `checkArgument()` if you think that's more appropriate.

Reply:
prefer check argument